### ISSUE TITLE
Possible memory leak when unmounting

### ISSUE THREAD

**Issue by bodo22** [hodgef/react-simple-keyboard#315]
When often unmounting and mounting the Keyboard component, I am experiencing a memory leak. A missing destroy() may be the culprite.

I am able to let the GC do its thing when adding the following to the Keyboard component

```
useEffect(() => {
  return () => {
    keyboardRef.current.destroy();
  };
}, []);
```

Here is a screenshot of a snapshot when not adding the .destroy() and remounting often (the keyboard is not the only thing mounted here):

![screenshot](https://user-images.githubusercontent.com/7796697/70223579-3dccb280-174c-11ea-8821-c24e808bf724.png)

**Comment by hodgef** [comment]
Hey @bodo22,

Thanks for reporting this. Indeed, `destroy` is only called on-demand, this keeps the behavior consistent with the [vanilla js simple-keyboard](https://github.com/hodgef/simple-keyboard).

In the core component, visibility toggling is [done with CSS](https://franciscohodge.com/projects/simple-keyboard/qa-use-cases/show-hide-toggle/), which keeps the same instance throughout the app's lifetime.

One reason not to call destroy automatically though, is an issue I've seen with routers when pages are transitioning. You'd get an instantiation happening at the same time a destroy command is being processed for the same target element. This happens in crossfade transitions when two pages are loaded at the same time to achieve the effect.

So long story short: I think for the time being I will keep `destroy` as an opt-in, something devs can call depending on the logic they have in place.

I will improve the [Troubleshooting section of the Q&A docs](https://franciscohodge.com/projects/simple-keyboard/qa-use-cases/) this week, to make sure people are aware of this to prevent memory issues.

Thanks again,
Francisco Hodge
